contracts/AgentMech.sol

```diff
@@ -76,9 +103,57 @@ error Overflow(uint256 provided, uint256 max);
 /// @dev Caught reentrancy violation.
 error ReentrancyGuard();
 
-/// @title AgentMech - Smart contract for extending ERC721Mech
+/// @dev Wrong state of a service.
+/// @param state Service state.
+/// @param serviceId Service Id.
+error WrongServiceState(uint256 state, uint256 serviceId);
+
+/**
+ * @dev A Mech that is operated by the multisig of an Olas service
+ */
+contract OlasMech is Mech, ImmutableStorage {
```

Comment:
This is almost identical to ERC721Mech except the constructor and isOperator implementations